ably/auth.go

```diff
@@ -127,7 +127,7 @@ func (a *Auth) clientIDForCheck() string {
 func (a *Auth) updateClientID(clientID string) {
 	a.mtx.Lock()
 	defer a.mtx.Unlock()
-	if a.clientID == "" {
+	if clientID != "" {
```

Comment:
Could you explain this change? In particular, this seems to preclude the possibility of the library being identified and then becoming unidentified. But [RSA12a] seems to require that this may happen.

(Not saying that the condition before is right. It seems wrong too, per RSA7b.)

Reply:
I think we can remove the check. We don't need to it in order to conform to both cases. The check shouldn't have been there from the start, since we need to set whatever value we received as clientId